cmd/src/batch_common.go

```diff
@@ -48,27 +49,39 @@ type batchExecuteFlags struct {
 	textOnly bool
 }
 
-func newBatchExecuteFlags(flagSet *flag.FlagSet, cacheDir, tempDir string) *batchExecuteFlags {
+func newBatchExecuteFlags(flagSet *flag.FlagSet, workspaceExecution bool, cacheDir, tempDir string) *batchExecuteFlags {
 	caf := &batchExecuteFlags{
 		api: api.NewFlags(flagSet),
 	}
 
-	flagSet.BoolVar(
-		&caf.textOnly, "text-only", false,
-		"INTERNAL USE ONLY. EXPERIMENTAL. Switches off the TUI to only print JSON lines.",
-	)
-	flagSet.BoolVar(
-		&caf.allowUnsupported, "allow-unsupported", false,
-		"Allow unsupported code hosts.",
-	)
-	flagSet.BoolVar(
-		&caf.allowIgnored, "force-override-ignore", false,
-		"Do not ignore repositories that have a .batchignore file.",
-	)
-	flagSet.BoolVar(
-		&caf.apply, "apply", false,
-		"Ignored.",
-	)
+	if !workspaceExecution {
+		flagSet.BoolVar(
+			&caf.textOnly, "text-only", false,
+			"INTERNAL USE ONLY. EXPERIMENTAL. Switches off the TUI to only print JSON lines.",
+		)
+		flagSet.BoolVar(
+			&caf.allowUnsupported, "allow-unsupported", false,
+			"Allow unsupported code hosts.",
+		)
+		flagSet.BoolVar(
+			&caf.allowIgnored, "force-override-ignore", false,
+			"Do not ignore repositories that have a .batchignore file.",
+		)
+		flagSet.BoolVar(
+			&caf.apply, "apply", false,
+			"Ignored.",
+		)
+		flagSet.BoolVar(
+			&caf.keepLogs, "keep-logs", false,
+			"Retain logs after executing steps.",
+		)
```

Comment:
technically this flag could be of use, if we ever allowed something like "ssh into container when failed" or something like that.

Reply:
But we do log more via the JSON log lines than we ever did in these log files. So I don't think they'd be useful or am I missing a usecase?

Reply:
Oh is this only stdout/stderr? Then nvm

Reply:
That and a bunch of other things, but not more useful than the JSON output: https://sourcegraph.com/search?q=context:%40sourcegraph/all+file:executor+Logf%28+repo:src-cli&patternType=literal

Reply:
great :)